### PR TITLE
kafka: synchronously insert events into kafka

### PR DESCRIPTION
**Pull request by vincentbernat**
This allows us to check for errors and avoid losing logs. However, this
makes the insertion quite slow.

The PR is WIP, I will investigate more about the performance issue. Moreover, if Kafka is unavailable, the thread will just wait forever. Maybe we should fail after some time instead?

cc @pyr

### CODE REVIEW

**Comment by vincentbernat** [comment]
With some help from @edenhill, the performance can be restored by setting the `queue.buffering.max.ms` property to `1` but this will still limit the overall performance to 1000 msg/s. I'll make the sync behaviour optional.

**Comment by vincentbernat** [comment]
So, the synchronous behavior is now disabled by default and should be enabled with the `flags(sync)` option:

```
    destination {
      kafka(flags(sync)
            properties(metadata.broker.list("localhost:9092")
                       queue.buffering.max.ms("1"))
            topic("logs")
            payload("$(format-json --scope nv-pairs --exclude SOURCE --exclude MESSAGE --exclude HOST* --exclude FILE_NAME)"));
    };
```

I still think this is a desirable feature to avoid logs to be just discarded on the first network outage.

**Comment by pyr** [comment]
I think it's a very nice feature as an option. :+1:

File: modules/kafka/kafka.c [hunk 1]
```diff
@@ -313,34 +348,59 @@ kafka_worker_insert(LogThrDestDriver *s, LogMessage *msg)
...
 
-  rd_kafka_produce(self->topic,
-                   RD_KAFKA_PARTITION_UA,
-                   RD_KAFKA_MSG_F_COPY,
-                   self->payload_str->str,
-                   self->payload_str->len,
-                   &key, sizeof(key),
-                   NULL);
-  msg_debug("Kafka produce done", 
+  rd_kafka_resp_err_t err = -12345;
```
Comment (juhaszviktor) [inline modules/kafka/kafka.c]: This -12345 should be a define, what do you think about INITIAL_ERROR_CODE?

**Comment by juhaszviktor** [comment]
Thank you, It looks good and useful patch. :+1:

File: modules/kafka/kafka.c (hunk 1 above)
Comment (vincentbernat) [inline modules/kafka/kafka.c]: OK, I have updated the patch.

**Comment by juhaszviktor** [comment]
:+1: Nice!

**Pull request merged**